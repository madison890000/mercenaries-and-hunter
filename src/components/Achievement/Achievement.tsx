import Tag from '../Tag';
import React from 'react';
import StringWithID from '../../models/StringWithID';
import capitalize from '../../utils/capitalize';
import pipe from '../../utils/pipe';
import { addPeriodSuffix } from '../../utils/suffix';

interface AchievementProps {
    title: string;
    categories: StringWithID[];
}

const Achievement = ({ title, categories }: AchievementProps) => (
    <li>
        {pipe<string>(capitalize, addPeriodSuffix)(title)}
        <span>
            {categories?.map(c => (
                <Tag key={c.id}>{c.toString()}</Tag>
            ))}
        </span>
    </li>
);

export default Achievement;
