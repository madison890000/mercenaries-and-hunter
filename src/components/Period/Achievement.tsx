import Tag from '../Tag';
import React from 'react';

interface AchievementProps {
    title: string;
    categories: string[];
}

const Achievement = ({ title, categories }: AchievementProps) => (
    <li>
        {title}
        <span>
            {categories?.map(c => (
                <Tag>{c}</Tag>
            ))}
        </span>
    </li>
);

export default Achievement;
