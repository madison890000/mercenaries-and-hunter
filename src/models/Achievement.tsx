import React from "react";
import Base from "./Base";
import EditText from "./EditText";

import {addPeriodSuffix, capitalize, pipe} from "../utils";
import Keywords from "./Keywords";

class Achievement extends Base {
    private categories: Keywords;
    text: EditText;

    constructor(text: string = '', categories: string[] = []) {
        super();
        this.text = new EditText(pipe<string>(capitalize, addPeriodSuffix)(text), 'input', '成就').setParent(this);
        this.categories = new Keywords(categories).setParent(this);
        this.showName = '成就';
    }

    View = () => {
        const Text = this.text.Show;
        const Categories = this.categories.Show;
        return (
            <li>
                <div style={{display: 'flex', alignItems: 'center'}}>
                    <Text/>
                    <span>
                    <Categories/>
                </span>
                </div>
            </li>
        )
    }

    updateTranslate(achievement: any) {
        this.text.text = achievement.text;
    }
}

export default Achievement