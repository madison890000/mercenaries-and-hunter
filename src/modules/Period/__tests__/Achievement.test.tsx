import React from 'react';
import Achievement from '../../../components/Achievement/Achievement';
import renderer from 'react-test-renderer';

test('renders Achievement', () => {
    const title = 'test Achievement';
    const categories = ['test category 1', 'Engineering'];
    const AchievementDom = renderer.create(<Achievement title={title} categories={categories} />).toJSON();
    expect(AchievementDom).toMatchSnapshot();
});
