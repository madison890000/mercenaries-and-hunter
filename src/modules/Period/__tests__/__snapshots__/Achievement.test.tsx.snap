// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Achievement 1`] = `
<li>
  Test Achievement.
  <span>
    <div
      className="normal tag"
    >
      test category 1
    </div>
    <div
      className="normal tag"
    >
      Engineering
    </div>
  </span>
</li>
`;
